tests/console/system_prepare.pm

```diff
@@ -34,8 +34,12 @@ sub run {
 
     prepare_serial_console;
 
-    # Make sure packagekit is not running, or it will conflict with SUSEConnect.
-    pkcon_quit unless check_var('DESKTOP', 'textmode');
+    if (!check_var('DESKTOP', 'textmode')) {
+        # Make sure packagekit is not running, or it will conflict with SUSEConnect.
+        pkcon_quit;
+        # poo#77134 wait for packagekit related zypper process to exit
+        assert_script_run 'until ! pgrep zypper; do sleep 1; done';
```

Comment:
can i suggest use of script_retry https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/master/lib/utils.pm#L1493

Reply:
Thanks, I think it is correct function to do such things here. I just think both work for me, I will try it next time. Thanks.

Reply:
I recommend against that as the recurring switch between the scope of remote-shell and local-perl is more expensive and error prone. `script_retry` can be helpful if something as simple as such loop does not work.

Reply:
Thanks for the analysis for the difference for the loop script and the function script_retry. It's very helpful.